orttraining/orttraining/test/distributed/partition_utils.hpp

```diff
@@ -0,0 +1,647 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
This header file is pretty heavy. Depending on what others think I would suggest to move the implementation of the non-template functions to a .cc file instead.

Reply:
There may be some code repetition between this file and the pipeline_transformer.cc w.r.t. auxiliary functions. I'll wait for feedback from the ORT team on how to proceed here.